cibuildwheel/backend.py

```diff
@@ -0,0 +1,296 @@
+import contextlib
+from pathlib import Path, PurePath
+from types import TracebackType
+from typing import Dict, Iterator, Optional, Sequence, Type
+
+from .logger import log
+from .options import BuildOptions
+from .platform_backend import PlatformBackend
+from .typing import PathOrStr, assert_never
+from .util import (
+    NonPlatformWheelError,
+    get_build_verbosity_extra_flags,
+    prepare_command,
+)
+from .virtualenv import VirtualEnv, VirtualEnvBase
+
+
+class BuilderBackend:
```

Comment:
I think this would be a lot simpler as a dataclass.

Reply:
I will need some time to think about what it would look like going this way.